### PR TITLE
Add CA2022 to docs

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #41290. 

cc @mpidash 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams.md) | [docs/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams?branch=pr-en-us-43219) |
| [docs/core/whats-new/dotnet-8/sdk.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/core/whats-new/dotnet-8/sdk.md) | [docs/core/whats-new/dotnet-8/sdk](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8/sdk?branch=pr-en-us-43219) |
| [docs/core/whats-new/dotnet-9/sdk.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/core/whats-new/dotnet-9/sdk.md) | [docs/core/whats-new/dotnet-9/sdk](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-9/sdk?branch=pr-en-us-43219) |
| [docs/fundamentals/code-analysis/overview.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/fundamentals/code-analysis/overview.md) | [docs/fundamentals/code-analysis/overview](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview?branch=pr-en-us-43219) |
| [docs/fundamentals/code-analysis/quality-rules/ca1872.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/fundamentals/code-analysis/quality-rules/ca1872.md) | [docs/fundamentals/code-analysis/quality-rules/ca1872](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1872?branch=pr-en-us-43219) |
| [docs/fundamentals/code-analysis/quality-rules/ca2002.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/fundamentals/code-analysis/quality-rules/ca2002.md) | [docs/fundamentals/code-analysis/quality-rules/ca2002](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2002?branch=pr-en-us-43219) |
| [docs/fundamentals/code-analysis/quality-rules/ca2022.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/fundamentals/code-analysis/quality-rules/ca2022.md) | [docs/fundamentals/code-analysis/quality-rules/ca2022](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2022?branch=pr-en-us-43219) |
| [docs/fundamentals/code-analysis/quality-rules/ca2264.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/fundamentals/code-analysis/quality-rules/ca2264.md) | [docs/fundamentals/code-analysis/quality-rules/ca2264](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2264?branch=pr-en-us-43219) |
| [docs/fundamentals/code-analysis/quality-rules/index.md](https://github.com/dotnet/docs/blob/a536a5e6e639bd535bb69916058a5deff0a1f20f/docs/fundamentals/code-analysis/quality-rules/index.md) | [docs/fundamentals/code-analysis/quality-rules/index](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/index?branch=pr-en-us-43219) |

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca2264.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ When a value that's known to never be null is passed to `ArgumentNullException.T
 
 ## Rule description
 
-`ArgumentNullException.ThrowIfNull` throws when the passed argument is `null`. Certain constructs like non-nullable structs (except for <xref:System.Nullable%601>), 'nameof()' expressions and 'new' expressions are known to never be null, so `ArgumentNullException.ThrowIfNull` will never throw.
+`ArgumentNullException.ThrowIfNull` throws when the passed argument is `null`. Certain constructs like non-nullable structs (except for <xref:System.Nullable%601>), 'nameof()' expressions, and 'new' expressions are known to never be null, so `ArgumentNullException.ThrowIfNull` will never throw.
```
Comment (BillWagner) [inline docs/fundamentals/code-analysis/quality-rules/ca2264.md:29]: Hmm. Open question here:  What generic constraints would trigger this warning?  Consider:

```csharp
public void M<T>(T arg) where T : struct
{
    ArgumentNullException.ThrowIfNull(...); // I'd expect CA2264 here.
}
```

But, what about:

```
public void M<T>(T arg) where T : notnull
public void M2<T>(T arg) where T : unmanaged
```

I'd expect CA2264 on both methods.
And, finally, I'd expect this wouldn't produce a warning:

```
public void M<T>(T arg) where T : allows ref struct
```

**Review by BillWagner — APPROVED** [review]
This looks great @gewarren 

I did have a question on how this rule gets handled with type parameters and constraints

File: docs/fundamentals/code-analysis/quality-rules/ca2264.md (hunk 1 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/quality-rules/ca2264.md:29]: @BillWagner What is the action item for me here - do you want me to expand the description, or test these out and then expand the description depending on what I find?

Comment (BillWagner) [reply docs/fundamentals/code-analysis/quality-rules/ca2264.md:29]: @gewarren 

What I wrote above was my guess at what that warning would do. If you've already got a test harness in place, can you test these?  The first three are the only ones I'd expect to trigger this warning. The final one shouldn't.

Comment (gewarren) [reply docs/fundamentals/code-analysis/quality-rules/ca2264.md:29]: Here's my results:

void M<T>(T arg) where T : struct
{
    ArgumentNullException.ThrowIfNull(arg); // CA2264
}
    
void M1<T>(T arg) where T : notnull
{
    ArgumentNullException.ThrowIfNull(arg); // No violation
}
    
void M2<T>(T arg) where T : unmanaged
{
    ArgumentNullException.ThrowIfNull(arg); // CA2264

}

void M3<T>(T arg) where T : allows ref struct
{
    ArgumentNullException.ThrowIfNull(arg); // CS1503
}

Comment (BillWagner) [reply docs/fundamentals/code-analysis/quality-rules/ca2264.md:29]: Here's what I'd change based on those results:

```suggestion
`ArgumentNullException.ThrowIfNull` throws when the passed argument is `null`. Certain constructs like non-nullable structs (except for <xref:System.Nullable%601>), type parameters known to be non-nullable structs, 'nameof()' expressions, and 'new' expressions are known to never be null, so `ArgumentNullException.ThrowIfNull` will never throw.
```

**Pull request merged**